Collections/Collections.tex

```diff
@@ -1135,6 +1134,71 @@ \section{Some hints for using collections}
+While an in-place-sort might reach higher performance in some situations, an out-of-place sort is the preferable option in most situations because \begin{enumerate*}[label={\alph*)}]\item it helps you avoid unintended side-effects that eventually can make your code harder to debug and understand\footnote{%
+    Actually, it is dangerous to modify an inline array or string from your code in-place (and thus forbidden since \sq 5.3).%
+} and \item it does not restrict the collection you use to be an array or an ordered collection\end{enumerate*}.
+
+The following examples illustrates the difference between in-place and out-of-place sorting:
+
+\begin{code}{@TEST |array|}
+array := {3. 5. 1}.
+
+"Out-of-place sort"
+array sorted. --> #(1 3 5)
+array. --> #(3 5 1)
+
+"In-place sort"
+array sort.
+array. --> #(1 3 5)
+\end{code}
+
+Sort methods use the message \mthind{Magnitude}{<=} to establish sort order, so by default, they can sort magnitudes such as numbers, characters, times \etc (see \charef{basic} on \ct{Magnitude}), but also other objects such as strings or points.
+However, what can you do if you want to sort objects that do not implement \ct{<=}, or if you would like to customize sort order?
+For this purpose, \ct{Collection} also provides \mthind{Collection}{sorted:} in addition to \ct{sorted} (note the colon), and analogously, arrays \etc also implement \mthind{ArrayCollection}{sort:}.
+These messages take a two-argument sort block or a \emphind{sort function}\index{SortFunction, class} that, speaking formally, defines a partial order on the collection to be sorted.%
+\footnote{Further reading: \href{https://en.wikipedia.org/wiki/Partially_ordered_set}{Wikipedia -- Partially ordered set}}
+Here is a simple example of how to use a sort block:
+
+\begin{code}{@TEST}
+#('Foo' 'Bar' 'baz') sorted: [:a :b | a caseInsensitiveLessOrEqual: b] --> #('Bar' 'baz' 'Foo')
+\end{code}
+
+In many cases, you will need to sort a collection by a particular property or transformation of each element.
+This is often realized by implementing the property function twice, e.g. (beware, negative example!):
+\begin{code}{@TEST}
+(1 to: 10) sorted: [:a :b | (a raisedTo: 2 modulo: 3) <= (b raisedTo: 2 modulo: 3)] --> #(3 6 9 1 2 4 5 7 8 10)
+\end{code}
+
+However, the sort function protocol provides a much more elegant solution to this problem by defining a dedicated \clsind{PropertySortFunction} class.
+It can be instantiated from any unary block by sending the message \mthind{BlockClosure}{ascending} or \mthind{BlockClosure}{descending} to it:
+
+\begin{code}{@TEST}
+(1 to: 10) sorted: [:ea | ea raisedTo: 2 modulo: 3] ascending --> #(3 6 9 1 2 4 5 7 8 10)
+\end{code}
+
+\mthind{Symbol}{ascending} and \mthind{BlockClosure}{descending} are defined on \clsind{Symbol} as well, so we could also write the previous example in the following way (but the former version is more efficient):
```

Comment:
The former version is also considered better style.

Reply:
```suggestion
\mthind{Symbol}{ascending} and \mthind{BlockClosure}{descending} are defined on \clsind{Symbol} as well, so we could also write the previous example in the following way (however, the former version is more efficient, and it also would be considered better style):
```